### PR TITLE
Incoming Channel Map w/ generated code

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers

### CODE REVIEW

File: shared/util/saga.js [hunk 1]
```diff
@@ -20,22 +20,26 @@ export function putOnChannelMap<T> (channelMap: ChannelMap<T>, k: string, v: T):
...
 
 // TODO type this properly
-export function takeFromChannelMap<T> (channelMap: ChannelMap<T>, k: string): any {
+export function effectOnChannelMap<T> (effect: any, channelMap: ChannelMap<T>, k: string): any {
   const c = channelMap[k]
   if (c) {
-    return take(c)
+    return effect(c)
   } else {
     console.error('Trying to take, but no registered channel for', k)
```
Comment (chrisnojima) [inline shared/util/saga.js]: this is no longer take

**Comment by chrisnojima** [comment]
👍 % 💬

**Comment by chromakode** [comment]
:+1:

**Comment by MarcoPolo** [comment]
I'm merging because of flaky tests

**Pull request merged**